application/forms/PreferenceForm.php

```diff
@@ -202,6 +209,46 @@ public function createElements(array $formData)
             }
         }
 
+        if (isset($formData['theme']) && $formData['theme'] !== StyleSheet::DEFAULT_THEME) {
+            $file = file_get_contents(StyleSheet::getThemeFile($formData['theme']));
+            if (! preg_match(StyleSheet::REGEX_ALL_MODE_QUERY, $file)) {
+                $disabled = ['', 'light', 'system'];
+                if (preg_match(StyleSheet::REGEX_AUTO_MODE_QUERY, $file)) {
+                    $value = 'system';
+                    $disabled = ['', 'light'];
+                }
+            }
+        }
+
+        $this->addElement(
+            'radio',
+            'theme_mode',
+            [
+                'class'         => 'theme-mode-input',
+                'label'         => $this->translate('Theme Mode'),
+                'multiOptions'  => [
+                    ''          => new HtmlElement(
+                        'img', [
+                                'src' => $this->getView()->href('img/theme-mode-thumbnail-dark.svg')
+                            ]
+                        ) . $this->translate('Dark'),
+                    'light'     => new HtmlElement(
+                        'img', [
+                                'src' => $this->getView()->href('img/theme-mode-thumbnail-light.svg')
+                            ]
+                        ) . $this->translate('Light'),
+                    'system'    => new HtmlElement(
+                        'img', [
+                            'src' => $this->getView()->href('img/theme-mode-thumbnail-system.svg')
+                            ]
+                        ) . $this->translate('System')
+                ],
+                'value'         => isset($value) ? $value : '',
+                'disable'       => isset($disabled) ? $disabled : [],
+                'escape' => false
```

Comment:
Woah, check your alignment. Also, use `HtmlElement::create()` instead of `new HtmlElement()`, the latter changed recently and it won't work anymore how you're using it here.

Reply:
✅